### PR TITLE
DevOps - Deploy joystream stack using docker-compose on EC2

### PR DESCRIPTION
**Pull request by ahhda**
### To run locally:
* Copy `deploy-single-node.sample.cfg` and update the `BRANCH_NAME` and `GIT_REPO` parameters
* Change directory to `devops/aws`
* Run `./deploy-stack-docker-compose.sh deploy-single-node.sample.cfg`

### To run Github Action
* Merge the Github Action to main branch of the repository
* Go to actions > Deploy docker-compose stack on AWS EC2
* Click on Run workflow > Add inputs
* Submit

Screenshot 2021-11-24 at 1 21 10 AM

Sample Github action successful run: https://github.com/ahhda/joystream/runs/4303647975?check_suite_focus=true

Warning: This script does not delete the instance once created. Deletion will have to be done manually by the user using the console or delete-stack script.

### CODE REVIEW

**Review by mnaamani — CHANGES_REQUESTED** [review]
I pushed an update so you can start persisted docker services now with:

`PERSIST=true yarn start`

I tried deploying however ansible step `yarn start` didn't seem to work correctly?

```
TASK [common : Delete remote code directory if exists] ******************************************************************************************************
ok: [54.174.212.175]

TASK [common : Git checkout] ********************************************************************************************************************************
changed: [54.174.212.175]

TASK [Creat bash profile file] ******************************************************************************************************************************
changed: [54.174.212.175]

TASK [Run setup script] *************************************************************************************************************************************
changed: [54.174.212.175]

TASK [Copy bash_profile content] ****************************************************************************************************************************
changed: [54.174.212.175]

TASK [Copy bash_profile content to bashrc for non-interactive sessions] *************************************************************************************
changed: [54.174.212.175]

TASK [Make sure docker is running] **************************************************************************************************************************
changed: [54.174.212.175]

TASK [Build packages] ***************************************************************************************************************************************
changed: [54.174.212.175]

TASK [Build Node image] *************************************************************************************************************************************
changed: [54.174.212.175]

TASK [Run docker-compose] ***********************************************************************************************************************************
changed: [54.174.212.175]

TASK [Set nip.io domain with IP] ****************************************************************************************************************************
ok: [54.174.212.175]

TASK [Install and configure Caddy] **************************************************************************************************************************

TASK [caddy_ansible.caddy_ansible : Assert usage of systemd as an init system] ******************************************************************************
ok: [54.174.212.175] => changed=false 
  msg: All assertions passed
```

**Comment by mnaamani** [comment]
Do we need to change working directory to the cloned repo directory?

File: devops/aws/templates/Caddyfile.j2 [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{{ nip_domain }}/ws-rpc {
+    reverse_proxy localhost:9944
```
Comment (mnaamani) [inline devops/aws/templates/Caddyfile.j2:3]: need to strip  /ws-rpc prefix

Comment (mnaamani) [reply devops/aws/templates/Caddyfile.j2:3]: for http-rpc, colossus and distributor nodes as well

File: devops/aws/templates/Caddyfile.j2 [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{{ nip_domain }}/colossus {
+    reverse_proxy localhost:3333
+}
+
+{{ nip_domain }}/distributor {
+    reverse_proxy localhost:3334
+}
+
+{{ nip_domain }}/graphql* {
+    uri strip_prefix /graphql
```
Comment (mnaamani) [inline devops/aws/templates/Caddyfile.j2]: do not strip prefix here

Comment (ahhda) [reply devops/aws/templates/Caddyfile.j2]: Done

File: devops/aws/templates/Caddyfile.j2 (hunk 1 above)
Comment (ahhda) [reply devops/aws/templates/Caddyfile.j2:3]: Done

**Comment by mnaamani** [comment]
Deploying works as expected from localhost. I'm having trouble seeing the Action show up to actually test.

**Pull request merged**